### PR TITLE
refactor: Accept parent process as argument to `proxySignals`

### PR DESCRIPTION
**Pull request by demurgos**
# Why

`foregroundChild` is currently hard-coded to only support forwarding with the root process. To support a general `proxy` function between any two objects behaving like processes, we can begin by refactoring the internals to be more parametrized.

# What

Document `proxySignals` and update its signature to accept the parent process.

### CODE REVIEW

**Comment by demurgos** [comment]
The CI failure seems to be an Appveyor issue (issue in their `npm` package). Fixed in https://github.com/tapjs/foreground-child/pull/24.

File: index.js [hunk 1]
```diff
@@ -83,20 +83,28 @@ module.exports = function (program, args, cb) {
...
   var listeners = {}
   signalExit.signals().forEach(function (sig) {
     listeners[sig] = function () {
       child.kill(sig)
     }
-    process.on(sig, listeners[sig])
+    parent.on(sig, listeners[sig])
   })
 
   return unproxySignals
```
Comment (profnandaa) [inline index.js]: Why not just return the function (with it's definition, could be named or anonymous):

```javascript
return function () {
  // ...
}
```

**Review by profnandaa — COMMENTED** [review]
Again, just nitpicks.

File: index.js (hunk 1 above)
Comment (demurgos) [reply index.js]: The focus of this PR was to pass the parent as input so I kept the other changes minimal. I can change it if you want to.

**Review by bcoe — APPROVED** [review]
if you have a moment I'd advocate making @ProfNandaa's change, but I won't block landing. Let me know what you decide and I'll merge.

**Comment by demurgos** [comment]
@bcoe I rebased my changes and applied @ProfNandaa's suggestion (return the function immediately)

**Review by profnandaa — APPROVED** [review]

**Comment by bcoe** [comment]
do we still want to land some of the work in `foreground-child` and `spawn-wrap`; haven't heard from you in a bit @demurgos.

**Comment by demurgos** [comment]
Hi,
I had less time at the beginning of the year and was absent. I am currently checking all the repos I watch and open issues. I also joind the `node-tooling` Slack and should be there more regularly.

Regarding this specific issue, I'd still like it to land. It's content is still good and self-contained.

I no longer use `spawn-wrap` but I'd be willing to help merging my fixes if there is interest for `istanbul`.

**Pull request closed without merging**